gate-oauth2/src/test/java/com/netflix/spinnaker/gate/security/oauth2/SpinnakerUserInfoTokenServicesTest.java

```diff
@@ -100,6 +106,27 @@ public void shouldEvaluateUserInfoRequirementsAgainstAuthenticationDetails() {
         .isFalse();
   }
 
+  @Test
+  public void verifyIsServiceAccount() {
+    PermissionService permissionService = mock(PermissionService.class);
+    when(permissionService.isEnabled()).thenReturn(true);
+
+    Front50Service front50Service = mock(Front50Service.class);
+    ServiceAccount serviceAccount = new ServiceAccount();
+    serviceAccount = serviceAccount.setName("ex@foo.com");
+    when(front50Service.getServiceAccounts()).thenReturn(Calls.response(List.of(serviceAccount)));
+
+    SpinnakerUserInfoTokenServices tokenServices = new SpinnakerUserInfoTokenServices();
+    tokenServices.userInfoMapping = new OAuth2SsoConfig.UserInfoMapping();
```

Comment:
This kind of direct member access isn't great.  I'd say it's finally time to switch SpinnakerUserInfoTokenServices to use constructor injection instead of member injection.  Then I believe all the members can be private final, and tests like these can do what they need do by passing constructor arguments.

Reply:
https://github.com/rahul-chekuri/gate/commit/418b4f2bdfcece801395a9c12f87a5eed2a43de2